Integrations/ThreatX/ThreatX.py

```diff
@@ -0,0 +1,601 @@
+        commands['query'].update(my_timeframe)
+
+    return http_request('/entities', commands)
+
+
+@logger
+def get_entity_risk(entity_id):
+    commands = {
+        'command': 'risk_changes',
+        'id': entity_id
+    }
+
+    return http_request('/entities', commands)
+
+
+@logger
+def get_entities_command():
+    entity_name = demisto.args().get('entity_name', None)
+    entity_id = demisto.args().get('entity_id', None)
+    entity_ip = demisto.args().get('entity_ip', None)
+    timeframe = demisto.args().get('timeframe', None)
+    results = get_entities(entity_name, entity_id, entity_ip, timeframe)
+    e_risk = None
+    e_id = None
+    md = []
+    ec = {}
+    dbot_scores = []
+    ip_enrich = []
+
+    for entity in results:
+        risk_score = 0
+
+        # Grab the entity risk so we can set the Dbot score for the Actor IPs
+        e_id = entity.get('id')
+
+        if e_id:
+            e_risk = get_entity_risk(e_id)
+
+        if isinstance(e_risk, list) and e_risk:
+            if isinstance(e_risk[-1], dict) and 'risk' in e_risk[-1]:
+                risk_score = e_risk[-1]['risk']
+
+        entity['risk'] = risk_score
+
+        iplist = []
+
+        if 'actors' in entity:
+            for actor in entity['actors']:
+                if 'ip_address' in actor:
+                    ipdot = pretty_ip(actor['ip_address'])
+                    iplist.append(ipdot)
+                    actor['ip_address'] = ipdot
+
+                try:
+                    actor['interval_time_start'] = pretty_time(actor['interval_time_start'])
+                except KeyError:
+                    pass
+
+                try:
+                    actor['interval_time_stop'] = pretty_time(actor['interval_time_stop'])
+                except KeyError:
+                    pass
+
+                try:
+                    actor['fingerprint']['last_seen'] = pretty_time(actor['fingerprint']['last_seen'])
+                except KeyError:
+                    pass
+
+                dbscore = set_dbot_score(risk_score)
+
+                dbot_scores.append({
+                    'Vendor': 'ThreatX',
+                    'Indicator': ipdot,
+                    'Type': 'ip',
+                    'Score': dbscore
+                })
+
+                if dbscore == 3:
+                    ip_enrich.append({
+                        'Address': ipdot,
+                        'Malicious': {
+                            'Vendor': 'ThreatX',
+                            'Description': 'ThreatX risk score is ' + str(risk_score)
+                        }
+                    })
+                else:
+                    ip_enrich.append({
+                        'Address': ipdot
+                    })
+
+        md.append({'Name': entity['codename'],
+                   'ID': e_id,
+                   'IP Addresses': ', '.join(iplist),
+                   'ThreatX Risk Score': risk_score
+                   })
+
+        ec.update({
```

Comment:
you are overriding the keys in ec, this will result with only the last for-run.
Threatx.Entity is the only thing that will not be ok since it's not a list of all elements.